wiki/python/ESTIMATOR_GUIDE.md

```diff
@@ -0,0 +1,833 @@
+# cuML Python Estimators Developer Guide
+
+This guide is meant to help developers follow the correct patterns when creating/modifying any cuML Estimator object and ensure a uniform cuML API.
+
+**Note:** This guide is long, because it includes internal details on how cuML manages input and output types for advanced use cases. But for the vast majority of estimators, the requirements are very simple and can follow the example patterns shown below in the [Quick Start Guide](#quick-start-guide).
+
+Start by reading the following:
+
+1. [SkLearn Estimator Docs](https://scikit-learn.org/stable/developers/develop.html)
+   1. cuML Estimator design follows Scikit-learn very closely. We will only cover portions where our design differs from this document
+   2. Pay close attention to these sections as these topics have caused pain points in the past:
+      1. [Instantiation](https://scikit-learn.org/stable/developers/develop.html#estimated-attributes)
+      2. [Estimated Attributes](https://scikit-learn.org/stable/developers/develop.html#estimated-attributes)
+      3. [`get_params` and `set_params`](https://scikit-learn.org/stable/developers/develop.html#estimated-attributes)
+      4. [cloning](https://scikit-learn.org/stable/developers/develop.html#estimated-attributes)
+2. [Scikit-learn's Docstring Guide](https://scikit-learn.org/stable/developers/contributing.html#guidelines-for-writing-documentation)
+   1. We follow the same guidelines for specifying array-like objects, array shapes, dtypes, and default values
+
+## Quick Start Guide
+
+At a high level, all cuML Estimators must:
+1. Inherit from `cuml.common.base.Base`
```

Comment:
Quick start guide is definitely helpful... Thank you!

Reply:
Great. Glad to hear.